browser/brave_adaptive_captcha/brave_adaptive_captcha_service_factory.cc

```diff
@@ -46,6 +49,8 @@ class CaptchaDelegate
       return false;
     }
     return coordinator->ShowAdaptiveCaptcha();
+#endif
+    return true;
```

Comment:
Since we're essentially making this function a no-op on Android, maybe it would be smarter to do this where the function is called instead.

Reply:
I think you may have misunderstood me. I was suggesting to move this `BUILDFLAG` to where we call `ShowScheduledCaptcha` rather than doing it here.

Reply:
A different way (if we didn't want to change the caller) would be like this:
```cpp
bool ShowScheduledCaptcha(const std::string& payment_id,
                            const std::string& captcha_id) override {
#if BUILDFLAG(IS_ANDROID)
    return true;
#endif
    // Because this is triggered from the adaptive captcha tooltip, this call
    // isn't associated with any particular `Browser` instance and we can use
    // the last active browser for this profile.
    auto* profile = Profile::FromBrowserContext(context_);
    auto* browser = chrome::FindTabbedBrowser(profile, false);
    if (!browser) {
      return false;
    }
    // ...
```
```

Reply:
@bsclifton that's exactly what i did in the initial commit but the issue we have here is that `CaptchaDelegate` is part of `namespace` so it's being called even if we don't pass the delegate for android. and it's failing at `chrome::FindTabbedBrowser(profile, false);` which is not available for android.

Reply:
If we want to go this route, just make the `#endif` an `#else` instead. There's no need to build that code on Android (and based on what you're saying, it sounds like those functions don't exist there anyway).

Reply:
Done.